alerting/src/test/kotlin/org/opensearch/alerting/util/clustermetricscoordinatortest/ClusterMetricsCoordinatorIT.kt

```diff
@@ -0,0 +1,209 @@
+                }
+            }
+        }
+        flag = true
+        assertTrue(flag)
+    }
+
+    fun `test frequency`() {
+        client().updateSettings("plugins.alerting.cluster_metrics.execution_frequency", "2m")
+        Thread.sleep(300000)
+
+        val response = getResponse()
+        val hits = getHits(response)
+        val docs = hits["hits"] as ArrayList<Map<String, Any>>
+        val times = mutableSetOf<String>()
+
+        for (doc in docs) {
+            val source = doc["_source"] as Map<String, Map<String, Any>>
+            val metricType = source.keys.first()
+            times.add(source[metricType]?.get("timestamp").toString())
+        }
+        logger.info("this is times $times")
+        val time1 = Instant.parse(times.elementAt(times.size - 1))
+        val time2 = Instant.parse(times.elementAt(times.size - 2))
+        val diff = time2.until(time1, ChronoUnit.MINUTES)
+        assertEquals(diff, 2)
+    }
+
+    fun `test update storage time to less than minimum storage time`() {
+        assertThrows(IllegalArgumentException::class.java) {
+            try {
+                client().updateSettings("plugins.alerting.cluster_metrics.metrics_history_max_age", "30s")
+            } catch (t: ResponseException) {
+                logger.info("LOG THE RESPONSE EXCEPTION OBJECT ${t.cause}")
+            }
+        }
+    }
+
+    fun `test update frequency to less than minimum frequency`() {
+        assertThrows(IllegalArgumentException::class.java) {
+            client().updateSettings("plugins.alerting.cluster_metrics.execution_frequency", "1ms")
+        }
+    }
+
+    fun `test when execution frequency greater than storage time`() {
+        assertThrows(IllegalArgumentException::class.java) {
+            client().updateSettings("plugins.alerting.cluster_metrics.metrics_history_max_age", "20m")
+            client().updateSettings("plugins.alerting.cluster_metrics.execution_frequency", "25m")
+        }
+    }
+
+    fun `test successful client setting update`() {
+        client().updateSettings("plugins.alerting.cluster_metrics.metrics_history_max_age", "400m")
+        client().updateSettings("plugins.alerting.cluster_metrics.execution_frequency", "12m")
+        val response = getSettings()
+        val persistentMap = response["persistent"] as Map<String, Any>
+        val executionFrequency = persistentMap["plugins.alerting.cluster_metrics.execution_frequency"].toString()
+        val storageTime = persistentMap["plugins.alerting.cluster_metrics.metrics_history_max_age"].toString()
+        assertEquals(executionFrequency, "12m")
+        assertEquals(storageTime, "400m")
+    }
+
+    @After
+    // Reset the settings back to default, delete the created index.
+    fun cleanup() {
+        // reset settings
+        client().updateSettings("plugins.alerting.cluster_metrics.metrics_history_max_age", "7d")
+        client().updateSettings("plugins.alerting.cluster_metrics.execution_frequency", "15m")
```

Comment:
Rather than pass in the strings `7d` and `15m` here, can you call the default constants?

Reply:
Done in a new commit. Added default constants to the AlertingSettings.kt class and called it here.